Update runtime to 25.08, include 128x128 icon and fix build issues

Zishan-Rahman opened the pull request.

Zishan-Rahman committed:
Update runtime to 23.08
from 22.08

Zishan-Rahman committed:
Fix metainfo-ancient validation error

Zishan-Rahman committed:
Add developer name and default type for screenshot

AsciiWolf commented:
bot, build

AsciiWolf commented:
I thought that 64x64 is still supported on Flathub as the minimal desktop icon size, but it looks like it is 128x128 now:
```
appstream-missing-icon-file: No icon was generated by appstream. A >=128px PNG or SVG needs to be installed correctly
```
Sadly, there is no 128x128 px icon available for this game.

merwok commented:
Would it be not ok to resize the 64x64 version (with integer scaling)?

Otherwise, this image could be used: https://salsa.debian.org/perl-team/modules/packages/frozen-bubble/-/raw/master/share/gfx/menu/fblogo.png

AsciiWolf commented:
> Would it be not ok to resize the 64x64 version (with integer scaling)?

Sure. Actually, Flathub already does this and you can use the already generated 128x128 px icon from the [Frozen Bubble Flathub page](https://flathub.org/apps/org.frozen_bubble.frozen-bubble).

AsciiWolf commented:
Please fix the icon issue and if possible update the runtime to `24.08`.

**edit:** [Here](https://github.com/user-attachments/assets/2a5cf1c6-bdfc-4f75-8d14-e9798c08405f) is the correct icon that you can use.

AsciiWolf commented:
@Zishan-Rahman Any chance you could take a look at this? Thanks!

Zishan-Rahman committed:
Update runtime to 24.08
from 23.08

Zishan-Rahman committed:
Rename "app-id" to just "id" in Flatpak manifest

Zishan-Rahman committed:
Add 128x128 icon from Flathub page to Flatpak repository

Co-authored-by: AsciiWolf <mail@asciiwolf.com>

Zishan-Rahman committed:
Incorporate 128x128 icon into Flatpak manifest

Zishan-Rahman commented:
> @Zishan-Rahman Any chance you could take a look at this? Thanks!

@AsciiWolf Sorry for the ping, and sorry for the massive delay. It's been ages since I worked on a Flathub repository. I got so busy with other things, and I continue to be busy, but I managed to find enough time to work on this PR after I saw your ping. Thank you so much for reminding me, I completely forgot about this. I just pushed some commits, including the icon you linked to, that should hopefully settle these build issues once and for all. I'll also rename the PR itself as well.

Zishan-Rahman commented:
Oh no!

bot, build

Zishan-Rahman commented:
```
FB: Running: eu-strip --remove-comment --reloc-debug-sections -f /work/builddir/files/lib/debug/lib/perl5/site_perl/5.36.0/aarch64-linux/auto/Locale/gettext/gettext.so.debug -F /app/lib/debug/lib/perl5/site_perl/5.36.0/aarch64-linux/auto/Locale/gettext/gettext.so.debug /work/builddir/files/lib/perl5/site_perl/5.36.0/aarch64-linux/auto/Locale/gettext/gettext.so
eu-strip: while opening '/work/builddir/files/lib/perl5/site_perl/5.36.0/aarch64-linux/auto/Locale/gettext/gettext.so': Permission denied
Error: module perl-file-slurp: Child process exited with code 1
```
:thinking:

Zishan-Rahman commented:
Does anyone here know why the above error is happening? I haven't quite been able to put my finger on it yet.

AsciiWolf commented:
Sorry, but I have no idea.

You could try updating the perl-gettext module or removing the `"no-debuginfo": true,` build-option (to be honest, I don't know how useful it is anyway).

AsciiWolf commented:
bot, build

AsciiWolf commented:
Oh, the problem is in the _perl-file-slurp_ module. So, maybe try updating this module to a more recent version?

The latest one seems to be [File-Slurp-9999.32](https://cpan.metacpan.org/authors/id/C/CA/CAPOEIRAB/File-Slurp-9999.32.tar.gz).

AsciiWolf commented:
Any chance the _perl-file-slurp_ module could be updated? And probably also consider updating the runtime version to `25.08`.

AsciiWolf commented:
Any progress?

AsciiWolf commented on frozen-bubble.appdata.xml:
```diff
@@ -4,10 +4,12 @@
 -->
-<application>
-  <id type="desktop">frozen-bubble.desktop</id>
+<component type="desktop">
```
```suggestion
<component type="desktop-application">
```

The `desktop` component type is deprecated.

AsciiWolf commented:
bot, build

Zishan-Rahman committed:
Change deprecated component type

Co-authored-by: AsciiWolf <mail@asciiwolf.com>

AsciiWolf commented:
I am currently working on a build fix and it starts looking good. One more patch and it should hopefully build fine. :) I will post a full diff here for you to include into this PR when I get it working (hopefully later today).

By the way, do not try to update to the 25.08 runtime - it has much more build issues. 24.08 is good enough for now. :)

Zishan-Rahman committed:
Update runtime to 25.08

As recommended by @AsciiWolf

Zishan-Rahman committed:
Revert "Update runtime to 25.08"

This reverts commit 1907661e4f2cb04c23cc1bcadd2cf573b80802e6.

Zishan-Rahman commented:
> I am currently working on a build fix and it starts looking good. One more patch and it should hopefully build fine. :) I will post a full diff here for you to include into this PR when I get it working (hopefully later today).
> 
> By the way, do not try to update to the 25.08 runtime - it has much more build issues. 24.08 is good enough for now. :)

Ah, I missed your message! I'm undoing my latest edit now, then :+1:

AsciiWolf committed:
Fix the build

Signed-off-by: Zishan Rahman <zishanfrahman@outlook.com>

AsciiWolf commented:
[0001-Fix-the-build.patch](https://github.com/user-attachments/files/24395299/0001-Fix-the-build.patch)

Here is the full patch. Tested and works fine:

<img width="690" height="567" alt="screenshot" src="https://github.com/user-attachments/assets/29c38911-9c83-4071-8f81-641a350afbb9" />

LAN multiplayer also works. Internet one does not because the original website/master server (frozen_bubble.org) is no longer valid and hosts a fake gambling site (!) now.

**edit:** You may also have to run `git submodule update --init --recursive --remote` manually after applying this patch to actually update the shared-modules. And do not forget to `git add` the two new patches (_perl-sdl-*.patch_ files).

Zishan-Rahman commented:
```
stripping /work/builddir/files/lib/perl5/site_perl/5.36.0/x86_64-linux/auto/Locale/gettext/gettext.so to /work/builddir/files/lib/debug/lib/perl5/site_perl/5.36.0/x86_64-linux/auto/Locale/gettext/gettext.so.debug
eu-strip: while opening '/work/builddir/files/lib/perl5/site_perl/5.36.0/x86_64-linux/auto/Locale/gettext/gettext.so': Permission denied
```

:thinking:

Zishan-Rahman commented:
Let's try it again.

bot, build org.frozen_bubble.frozen-bubble

AsciiWolf commented:
Still the same perl-file-slurp module. There is no such problem on my local machine and I don't understand why it tries to run eu-elfcompress on and strip the debug symbols from the gettext.so library during the completely different module build process. Seems like a build tooling (or the perl-file-slurp Makefile) bug to me.

AsciiWolf commented:
This is probably an aarch64-specific issue in the perl-file-slurp (or perl-gettext) module build system. Please try the following:

```
diff --git a/org.frozen_bubble.frozen-bubble.json b/org.frozen_bubble.frozen-bubble.json
index fc958e1..6455047 100644
--- a/org.frozen_bubble.frozen-bubble.json
+++ b/org.frozen_bubble.frozen-bubble.json
@@ -220,11 +220,15 @@
         {
             "name": "perl-file-slurp",
             "no-autogen": true,
+            "build-options": {
+                "no-debuginfo": true,
+                "strip": false
+            },
             "sources": [
                 {
                     "type": "archive",
-                    "url": "https://search.cpan.org/CPAN/authors/id/C/CA/CAPOEIRAB/File-Slurp-9999.32.tar.gz",
-                    "sha256": "4c3c21992a9d42be3a79dd74a3c83d27d38057269d65509a2f555ea0fb2bc5b0"
+                    "url": "https://search.cpan.org/CPAN/authors/id/U/UR/URI/File-Slurp-9999.19.tar.gz",
+                    "sha256": "ce29ebe995097ebd6e9bc03284714cdfa0c46dc94f6b14a56980747ea3253643"
                 },
                 {
                     "type": "file",
```

(Note that I reverted it back to the original File-Slurp version. There is no need to update it to a newer fork with new potential issues if it does not fix the original problem.)

Not sure if `strip` is supported per-module though. If not, we will have to come with a different workaround. Disabling aarch64 would probably also work, but does not look like a good solution to me.

AsciiWolf committed:
Downgrade perl-file-slurp module to 9999.19
from 9999.32
in order to (hopefully) prevent further build errors

AsciiWolf commented:
It worked! We now have the same error, but on a different module: perl-ipc-system-simple

The same workaround should work there as well. :-)

Zishan-Rahman committed:
Implement same workaround in perl-file-slurp module for perl-ipc-system-simple module

Co-authored-by: AsciiWolf <mail@asciiwolf.com>

Zishan-Rahman committed:
Implement '"strip": false' workaround for perl-locale-maketext-lexicon module

Co-authored-by: AsciiWolf <mail@asciiwolf.com>

Zishan-Rahman committed:
Implement '"strip": false' workaround for perl-archive-zip module

Co-authored-by: AsciiWolf <mail@asciiwolf.com>

Zishan-Rahman commented:
bot, build

AsciiWolf commented:
We should probably consider adding `"no-debuginfo": true` and `"strip": false` globally for all modules. :)

Zishan-Rahman commented:
The thing is, this time we need to put `"no=debuginfo": true` and `"strip": false` in a _shared module_ (the SDL1 one), and since it's in a submodule, I can't just go in there and change it directly without changing the whole thing for the repo we're using. I'm not quite sure what the best approach would be here, so I'm going to summon the Flathub admins.

bot, ping admins

AsciiWolf commented:
Oh, oops. I did not notice it was in a submodule.

bbhtt commented:
you have to copy the manifest from shared modules into here ie. stop using the submodule. there is no other way.

AsciiWolf committed:
Disable debuginfo and debug symbols stripping globally

Signed-off-by: Zishan Rahman <zishanfrahman@outlook.com>

AsciiWolf commented:
[0001-Disable-debuginfo-and-debug-symbols-stripping-global.patch](https://github.com/user-attachments/files/24401387/0001-Disable-debuginfo-and-debug-symbols-stripping-global.patch)

Here is a better fix. :)

Tested and works fine: https://github.com/flathub/org.frozen_bubble.frozen-bubble/pull/13

hfiguiere commented:
maybe start by using the latest runtime. 25.08 is the current. And fixup permissions in the post install which is the root of the problem. It's not SDL1 the problem. It's some Perl stuff.

```
2025-12-31T23:07:56.0580806Z eu-strip: while opening '/work/builddir/files/lib/perl5/site_perl/5.36.0/aarch64-linux/auto/Compress/Bzip2/Bzip2.so': Permission denied
```

AsciiWolf commented:
I have already tried updating it to 25.08 before (see my previous comments), but there were multiple new build failures and I sadly do not have enough time to work on this properly at the moment. I would prefer if we could have a working 24.08 build for now.

I have also already tried fixing the permissions in post-install and it did not work for some reason. However, my current patch from [this comment](https://github.com/flathub/org.frozen_bubble.frozen-bubble/pull/12#issuecomment-3703770984) seems to work fine (the only drawback being that it disables debuginfo for all modules).

Zishan-Rahman commented:
When I get around to it, I'll get the SDL modules from the shared-modules repo directly onto this one. Disabling debug info for all modules should be fine as well. I can hopefully try updating the runtime again with @AsciiWolf's fixes as well. You'll know when I've got round to it.

AsciiWolf commented:
> I'll get the SDL modules from the shared-modules repo directly onto this one.

You won't need to add the SDL module if you use my patch. :)

Zishan-Rahman commented:
> > I'll get the SDL modules from the shared-modules repo directly onto this one.
> 
> You won't need to add the SDL module if you use my patch. :)

Note taken and patch applied :+1:

hfiguiere commented on org.frozen_bubble.frozen-bubble.json:
```diff
@@ -54,7 +58,7 @@
                 {
                     "type": "archive",
-                    "url": "http://www.cpan.org/src/5.0/perl-5.36.0.tar.gz",
+                    "url": "https://www.cpan.org/src/5.0/perl-5.36.0.tar.gz",
```
you should resync with the version that is in the SDK. I'm pretty sure it's more recent.

hfiguiere commented:
Disabling strip and debug isn't the solution.

> And fixup permissions in the post install which is the root of the problem. It's not SDL1 the problem. It's some Perl stuff.

But I guess ignoring the solution I propose...

Adding this in the Perl modules that cause the error seems to work.

```json
            "post-install": [
                "find /app/lib/perl5 -type f -exec chmod u+w {} \\;"
            ],
```

AsciiWolf commented:
> Adding this in the Perl modules that cause the error seems to work.

Thanks! I will try this and also the newer Perl version.

hfiguiere commented:
And frozen-bubble needs this:
```
                "find /app/lib/perl5 -type f -exec chmod u+w {} \\;",
                "chmod u+w $FLATPAK_DEST/bin/fb-server",
                "chmod u+w $FLATPAK_DEST/bin/frozen-bubble*",
```

Also frozen-bubble fail to build with perl 5.42

AsciiWolf commented:
Thanks for your help with this, Hubert!

Here is the permission fix patch:

[0001-Properly-workaround-the-permission-issues.patch](https://github.com/user-attachments/files/24403187/0001-Properly-workaround-the-permission-issues.patch)

I will now work on the Perl update.

AsciiWolf commented:
[0001-Update-Perl-to-5.38.5.patch](https://github.com/user-attachments/files/24403268/0001-Update-Perl-to-5.38.5.patch)

Last Perl version that Frozen Bubble builds fine with.

Tested and seems to work fine:

<img width="690" height="567" alt="screenshot" src="https://github.com/user-attachments/assets/71a3a3b1-4921-4514-b767-c32da6b1f02b" />

LAN multiplayer also works.

hfiguiere commented:
25.08? I could build it on 25.08. Not even need a new patch.

AsciiWolf committed:
Final fix

- Properly workaround the permission issues
- Update Perl to 5.38.5
- Update runtime to 25.08

Signed-off-by: Zishan Rahman <zishanfrahman@outlook.com>

AsciiWolf commented:
You are right! It builds and works fine with 25.08 now. :-)

Here is a new patch (feel free to ignore the previous two patches, one of them is not correct anyway):

[0001-Final-fix.patch](https://github.com/user-attachments/files/24403596/0001-Final-fix.patch)

AsciiWolf commented:
```
> ✅ [Test build succeeded](https://github.com/flathub-infra/vorarbeiter/actions/runs/20654078837).
```

<img width="690" height="567" alt="screenshot" src="https://github.com/user-attachments/assets/f5e7161c-d65a-4763-9a19-d0033089575c" />

Works fine. :)

AsciiWolf commented:
@hfiguiere I think this is now ready to be merged. :-)

(Just don't forget to squash the commits.)

Zishan-Rahman commented:
Nearly one-and-a-half years in the making! I think this PR's merging calls for a celebration, everyone! :tada:
